PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:4 + add:0, change matches:
-    _inherit = ['hr.employee']
-    _inherit = ['hr.resume.line']
-    _inherit = ['slide.channel.partner']
-    _inherit = ['slide.channel']

Total Changes: 8

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_skills_slides/models/hr_employee.py
+++ b/addons/hr_skills_slides/models/hr_employee.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models, _
 
 
 class HrEmployee(models.Model):
-    _inherit = ['hr.employee']
+    _inherit = 'hr.employee'
 
     subscribed_courses = fields.Many2many('slide.channel', related='user_partner_id.slide_channel_ids')
     has_subscribed_courses = fields.Boolean(compute='_compute_courses_completion_text', groups="hr.group_hr_user,base.group_system")

--- a/addons/hr_skills_slides/models/hr_resume_line.py
+++ b/addons/hr_skills_slides/models/hr_resume_line.py
@@ -5,7 +5,7 @@ from odoo import fields, models, api
 
 
 class HrResumeLine(models.Model):
-    _inherit = ['hr.resume.line']
+    _inherit = 'hr.resume.line'
 
     display_type = fields.Selection(selection_add=[('course', 'Course')])
     channel_id = fields.Many2one('slide.channel', string="Course", readonly=True, index='btree_not_null')

--- a/addons/hr_skills_slides/models/slide_channel.py
+++ b/addons/hr_skills_slides/models/slide_channel.py
@@ -8,7 +8,7 @@ from odoo.tools import html2plaintext
 
 
 class SlideChannelPartner(models.Model):
-    _inherit = ['slide.channel.partner']
+    _inherit = 'slide.channel.partner'
 
     def _recompute_completion(self):
         res = super(SlideChannelPartner, self)._recompute_completion()
@@ -66,7 +66,7 @@ class SlideChannelPartner(models.Model):
 
 
 class SlideChannel(models.Model):
-    _inherit = ['slide.channel']
+    _inherit = 'slide.channel'
 
     def _action_add_members(self, target_partners, member_status='joined', raise_on_access=False):
         res = super()._action_add_members(target_partners, member_status=member_status, raise_on_access=raise_on_access)
